### ISSUE TITLE
Properties Editor - Grease Pencil - Layers panel move layer up down into groups with arrows

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4652]
# BFA 4.1.1
![image](https://github.com/user-attachments/assets/4562f639-f922-4ba5-a480-db62aeb7f9dc)

# BFA master
![image](https://github.com/user-attachments/assets/fa307c01-be39-431d-ae0b-01143ffbda55)

# Tasks:
- Add layer up/down buttons

<details>
<summary>Details</summary>

SHA-1: c07c9d5729c82713281b860fc1998f7df3d951e7

* GPv3: Change "use_locked_material" option

This option is meant to ignore any material locking when it
comes to editing strokes that might use locked materials.

There were some issues with the current implementation.
* The name did not reflect what it was supposed to do, so it
   was renamed to `ignore_locked_materials`.
* The description of the option has been updated to better
   reflect the behavior.
* Some util functions have been refactored

Pull Request: https://projects.blender.org/blender/blender/pulls/127423

SHA-1: 9311a4c89cf7d407df828d3e195e30b4f8faa5b1

* GPv3: Layers menu in topbar

Add layers panel in topbar which consists list of layers
and operators such as add, hide, isolate, etc.

Resolves #123397.

Pull Request: https://projects.blender.org/blender/blender/pulls/127612

</details>

**Comment by Draise14** [comment]
I got the method to move layers up and down, but not into groups or out of groups. Also can't move the groups with the arrows yet, I assume this will be necessary.

**Comment by ReinerBforartists** [comment]
May i also remind here to just do one task at a time? You have defined three tasks here already, which should ring the alarm bells.

I know it is intimidating, but can lead to trouble when you solve too much steps at once. Like here, you have already solved problem one. But can't continue since you want to solve problem two too.

**Comment by Draise14** [comment]
Yep. Thought it would be simple. It's not. Will split after work, currently tied up. 

I updated the description to one task.

**Comment by Draise14** [comment]
- Added layer toggles to be the same
- Added move up and down, clarified the tooltips
- Cleaned up some grease pencil menus

**Note:** These don't move groups nor move layers into or out of groups, they seem to be a weird system with another indice, so hard or no way to know what position is the collection. I am not sure how.. I could build a way to move "out" but the index of where it would move out is unpredictable (ei, to top or bottom). In this case new operators to "move out of group" or "Move layer into new group" could be designed. I can also detect names of groups and their own position indices, but they are.. drawn some other way. API feels limited there. I left a comment about it.

**Note:** No point adding these operators to the 3D View layers panel, since the list is.. yet again, drawn differently and closes on single use. We would need new operators that will move layers up and down.

Will make a new task for that.

**Comment by Draise14** [comment]
Ah, now I think I see how it was done, a "dummy" socket was built into each group to have the group have an index in the node editor UI list. That was you @Blenux ? Clever! Took me most my day to understand. 

So... 

1. Create a "dummy" layer in each group
2. Name the dummy the layer name, now the group has an index
3. When you move down or up, the layer to a layer with the group layer name, then move into the "group" of that name
4. Then continue, if it can no-longer move in the group, move out and move to the index of the group dummy but one below. 
5. Then when you're done, remove the dummy layers to "untag" the groups. 

And then do this both ways. 

I guess I could make another task to move in and out of groups.

**Comment by Draise14** [comment]
![image](https://github.com/user-attachments/assets/b5c5c360-4fa8-4de3-b39c-1c0470561f13)

omfg, consistently inconsistent. I assumed they wouldn't add this due to the node editor groups ui list not having them. 

 They DID add the move up and down operators, but they still don't work moving in and out of groups.

**Comment by Draise14** [comment]
Will do a merge then try again.

**Comment by ReinerBforartists** [comment]
Merge is always good ^^

**Comment by Draise14** [comment]
API has improved, might make this easier. 
<details>
<summary>Details</summary>

SHA-1: 0a560582b587c5dcf9f0f942e88c2d385b8727c7

* Fix: GPv3: Python: Missing `parent_group` property on layer groups

This property was available for `layers` but was missed for layer groups.
Adds a `parent_group` property for layer groups that returns the
parent group if the group is inside another group.
For groups that are inside the root, returns `None`.

Pull Request: https://projects.blender.org/blender/blender/pulls/129168

</details>
